rts/Sim/MoveTypes/GroundMoveType.cpp

```diff
@@ -991,7 +990,7 @@ void CGroundMoveType::StopMoving(bool callScript, bool hardStop, bool cancelRaw)
 	LOG_L(L_DEBUG, "[%s] stopping engine for unit %i", __func__, owner->id);
 
 	if (!atGoal)
-		goalPos = (currWayPoint = Here());
+		earlyCurrWayPoint = (goalPos = (currWayPoint = Here()));
```

Comment:
That's becoming a fairly long chain of assignments. It's fine though and probably still more readable than the alternatives.